src/code_file_manager.rb

```diff
@@ -1,8 +1,9 @@
+require 'tempfile'
+
 module CodeFileManager
-  def next_file_absolute_path
-    possible_letters = [('a'..'z'), ('A'..'Z')].map(&:to_a).flatten
-    random_string = (0..10).map { possible_letters[rand(possible_letters.size)] }.join
-    "#{Dir.pwd}/codes/#{random_string}"
+  def next_file_absolute_path(extension)
+    tempfile = Tempfile.create(['foo', extension], "#{Dir.pwd}/codes")
```

Comment:
For example, Tempfile.create(['foo'. '.cpp']) creates

foo-212122-aas.cpp
foo-232112-dsd.cpp

for instance and each time its called it does it with a different name starting with foo and ending with .cpp. The library ensures that no file with the same file is created